python/lsst/ip/isr/linearize.py

```diff
@@ -26,226 +26,510 @@
+
+        Raises
+        ------
+        RuntimeError:
+            Raised if the linearity type listed in the
+            detector does not match the class type.
         """
-        ampInfoType = detector.getAmplifiers()[0].getLinearityType()
-        if self.LinearityType != ampInfoType:
-            raise RuntimeError("Linearity types don't match: %s != %s" % (self.LinearityType, ampInfoType))
+        pass
 
 
 class LinearizeLookupTable(LinearizeBase):
-    """Correct non-linearity with a persisted lookup table
+    """Correct non-linearity with a persisted lookup table.
 
     for each i,j of image:
         rowInd = int(c0)
         colInd = int(c1 + uncorrImage[i,j])
         corrImage[i,j] = uncorrImage[i,j] + table[rowInd, colInd]
 
-    where c0, c1 are collimation coefficients from the AmpInfoTable of the detector:
-    - c0: row index; used to identify which row of the table to use (typically one per amplifier,
-            though one can have multiple amplifiers use the same table)
-    - c1: column index offset; added to the uncorrected image value before truncation;
-            this supports tables that can handle negative image values; also, if the c1 ends with .5
-            then the nearest index is used instead of truncating to the next smaller index
+    where c0, c1 are collimation coefficients from the detector amplifier:
+    - c0: row index; used to identify which row of the table to use
+            (typically one per amplifier, though one can have multiple
+            amplifiers use the same table)
+    - c1: column index offset; added to the uncorrected image value
+            before truncation; this supports tables that can handle
+            negative image values; also, if the c1 ends with .5 then
+            the nearest index is used instead of truncating to the
+            next smaller index
 
-    In order to keep related data together, the coefficients are persisted along with the table.
-    """
-    LinearityType = "LookupTable"
+    In order to keep related data together, the coefficients are
+    persisted along with the table.
 
-    def __init__(self, table, detector):
-        """Construct a LinearizeLookupTable
-
-        @param[in] table  lookup table; a 2-dimensional array of floats:
-            - one row for each row index (value of coef[0] in the amp info catalog)
+    Parameters
+    ----------
+    table : `numpy.array`
+        Lookup table; a 2-dimensional array of floats:
+            - one row for each row index (value of coef[0] in the amplifier)
             - one column for each image value
-            To avoid copying the table the last index should vary fastest (numpy default "C" order)
-        @param[in] detector  detector information (an instance of lsst::afw::cameraGeom::Detector);
-            the name, serial, and amplifier linearization type and coefficients are saved
+        To avoid copying the table the last index should vary fastest
```

Comment:
What does this mean?

Reply:
The implementation works best if each row vector is contiguous in memory.  This is indexed by  (imVal + c1) in the previous comment.